dbuild/__init__.py

```diff
@@ -0,0 +1,174 @@
+    c = docker_client(docker_url)
+
+    docker_path=create_docker_dir(flavor, dist)
+
+    print "Starting %s Package Build" % build_type
+    image_tag='dbuild-' + flavor + '/' + dist
+    response = build_image(c, docker_path, tag=image_tag )
+    print response
+
+    if build_type == 'source':
+        command = ['dpkg-buildpackage', '-S', '-nc', '-uc', '-us']
+        cwd = '/build/' + source_dir
+    elif build_type == 'binary':
+        command = ['bash', '-c', "dpkg-source -x /build/*.dsc /build/pkgbuild/ && cd /build/pkgbuild && /usr/lib/pbuilder/pbuilder-satisfydepends && dpkg-buildpackage"]
+        cwd = '/build'
+    else:
+        shutil.rmtree(docker_path)
+        raise exceptions.DbuildBuildFailedException('Unknown build_type: %s' % build_type)
+
+    container = create_container(c, image_tag,command=command, shared_volumes={build_dir: '/build'}, cwd=cwd)
+    print(container)
+    response = start_container(c, container)
+    rv = wait_container(c, container)
+    logs = container_logs(c, container)
+    print '\n'.join(logs)
+
+    if rv == 0:
+        print 'Build successful (build type: %s), removing container %s' % (build_type, container.get('Id'))
+        remove_container(c, container, force=True)
+        build_rv = True
+    else:
+        if force_rm:
+            print "Build failed (build type: %s), Removing container %s" % (build_type, container.get('Id'))
+            remove_container(c, container, force=True)
+            build_rv = False
+        else:
+            print "Build failed (build type: %s), keeping container %s" % (build_type, container.get('Id'))
+            build_rv = False
+
+    shutil.rmtree(docker_path)
+    if build_rv:
+        return build_rv
+    elif build_type == 'source':
+        raise exceptions.DbuildSourceBuildFailedException('Source build FAILED')
+    elif build_type == 'binary':
+        raise exceptions.DbuildBinaryBuildFailedException('Binary build FAILED')
+
+def main(argv=sys.argv):
+    ap = argparse.ArgumentParser(description='Build debian packages in docker container')
+    ap.add_argument('--build_dir', type=str, help='package build directory')
+    ap.add_argument('--source_dir', type=str, default='source',
+            help='subdirectory of build_dir where sources kept')
+    ap.add_argument('--force_rm', action='store_true', default=False,
+            help='Remove the containers even if build failed')
+    ap.add_argument('--docker_url', type=str,
+            default='unix://var/run/docker.sock',
+            help='Docker url, it can be unix socket or tcp url')
+    ap.add_argument('--flavor', type=str, default='ubuntu',
+            help='Linux flavor to use for container to build')
+    ap.add_argument('--dist', type=str, default='trusty',
+            help='Linux distribution')
+
+    args = ap.parse_args()
+
+    if (not args.build_dir):
+        print 'build_dir must be provided'
+        ap.print_help()
+        sys.exit(1)
+
```

Comment:
If it's a required argument, you should make it positional. Just make it add_argument('build_dir') rather than add_argument('--build_dir').